### PR TITLE
[C#] feat: Switch tokenizer to use the Microsoft.ML.Tokenizers library

### LINKED ISSUE #1467: [C#]: Switch tokenizers to use Microsoft.ML.Tokenizers package
### Language

C#

### Version

latest

### Description

The .NET libraries teams have introduced the Microsoft.ML.Tokenizers library, incorporating the Tiktoken tokenizer utilized in the teams AI. Transitioning to this library offers the advantage of receiving comprehensive support akin to any .NET Microsoft library, alongside performance optimizations in terms of speed and memory allocations.

### PR DESCRIPTION
**Pull request by tarekgh**
fixes #1467

## Details

Switch to use Microsoft.ML.tokenizers library for tokenizer support.

#### Change details

> Describe your changes, with screenshots and code snippets as appropriate

**code snippets**:

**screenshots**:

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/Application/Authentication/AdaptiveCards/AdaptiveCardsAuthenticationBase.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ internal abstract class AdaptiveCardsAuthenticationBase
         /// </summary>
         /// <param name="context">The turn context</param>
         /// <returns>The sign in response</returns>
-        public async Task<SignInResponse> AuthenticateAsync(ITurnContext context)
+        public Task<SignInResponse> AuthenticateAsync(ITurnContext context)
```
Comment (singhk97) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/Application/Authentication/AdaptiveCards/AdaptiveCardsAuthenticationBase.cs:16]: could you undo this change as it's not related to the PR

Comment (tarekgh) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/Application/Authentication/AdaptiveCards/AdaptiveCardsAuthenticationBase.cs:16]: sure. I'll do that.

Comment (tarekgh) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/Application/Authentication/AdaptiveCards/AdaptiveCardsAuthenticationBase.cs:16]: Just FYI, this is producing a warning but I'll revert this change anyway.

**Review by singhk97 — APPROVED** [review]
